### PR TITLE
Fetch the hotshot liveness instead of inferring it

### PR DESCRIPTION
**Pull request by ImJeremyHe**
### This PR:
Validators will not infer the hotshot liveness from the messages. It would fetch the hotshot liveness from the hotshot light client contract.
This provides the STF the correct information to check the hotshot liveness.

### CODE REVIEW

File: staker/block_validator.go [hunk 1]
```diff
@@ -585,7 +585,10 @@ func (v *BlockValidator) createNextValidationEntry(ctx context.Context) (bool, e
 		return false, fmt.Errorf("illegal batch msg count %d pos %d batch %d", v.nextCreateBatchMsgCount, pos, endGS.Batch)
 	}
 	var comm espressoTypes.Commitment
-	var isHotShotLive bool
+	isHotShotLive, err := v.lightClientReader.IsHotShotLive(msg.Message.Header.BlockNumber)
+	if err != nil {
+		return false, fmt.Errorf("error fetching the hotshot liveness")
```
Comment (Sneh1999) [inline staker/block_validator.go:590]: nit: I think we should also attach the error here like  - `fmt.Errorf("error fetching the hotshot liveness: %v", err)`

**Review by Sneh1999 — COMMENTED** [review]
Also what happens if we cant fetch the liveness - should we just assume that `Hotshot` is down? and still proceed with transactions

File: staker/block_validator.go (hunk 1 above)
Comment (nomaxg) [reply staker/block_validator.go:590]: also attach the block number

**Comment by ImJeremyHe** [comment]
> Also what happens if we cant fetch the liveness - should we just assume that `Hotshot` is down? and still proceed with transactions

For now, the validators will just retry it again and again since there is a periodic task. The hotshot livenesses are stored in the L1 so I think there should be a large probability that we failed to get it.

**Review by Sneh1999 — APPROVED** [review]

**Pull request closed without merging**